### PR TITLE
[query] do not reinitialize zstd repeatedly

### PR DESCRIPTION
**Pull request by danking**
In the end, this did not seem to have any significant effect on decoding, but it does show up in the profiler as a hotspot.

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/InputBuffers.scala [hunk 1]
```diff
@@ -646,16 +652,15 @@ final class ZstdInputBlockBuffer(blockSize: Int, in: InputBlockBuffer) extends I
     } else {
       val compLen = blockLen - 4
       val decompLen = Memory.loadInt(comp, 0)
-      val ret = Zstd.decompressByteArray(buf, 0, decompLen, comp, 4, compLen)
-      if (Zstd.isError(ret))
-        throw new com.github.luben.zstd.ZstdException(ret)
+      val ret = zstd.decompressByteArray(buf, 0, decompLen, comp, 4, compLen)
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/io/InputBuffers.scala:655]: If we trust these methods to throw then we don't need to bind their return values.

File: hail/src/main/scala/is/hail/io/InputBuffers.scala [hunk 2]
```diff
@@ -675,9 +680,7 @@ final class ZstdSizedBasedInputBlockBuffer(blockSize: Int, in: InputBlockBuffer)
         compLen
       } else {
         val decompLen = decomp >>> 1
-        val ret = Zstd.decompressByteArray(buf, 0, decompLen, comp, 4, compLen)
-        if (Zstd.isError(ret))
-          throw new com.github.luben.zstd.ZstdException(ret)
+        val ret = zstd.decompressByteArray(buf, 0, decompLen, comp, 4, compLen)
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/io/InputBuffers.scala:683]: unused

**Review by chrisvittal — CHANGES_REQUESTED** [review]
Looks good. Thanks for making this change.

File: hail/src/main/scala/is/hail/io/InputBuffers.scala (hunk 1 above)
Comment (danking) [reply hail/src/main/scala/is/hail/io/InputBuffers.scala:655]: good point! For the record, they can fail, but the `throw` the exception. The return value is always the number of bytes, never an error code.

**Review by chrisvittal — APPROVED** [review]

**Pull request merged**